src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2492,6 +2530,23 @@ private void VisitArgumentEvaluate(ImmutableArray<BoundExpression> arguments, Im
             Debug.Assert(annotations.Length == arguments.Length);
             Debug.Assert(_disableDiagnostics);
 
+            // prepare slots for expressions and sub-expressions that we'll want to mark as non-null
+            var allSlots = ArrayBuilder<ArrayBuilder<int>>.GetInstance(arguments.Length);
```

Comment:
>allSlots [](start = 16, length = 8)

Do we need to hold onto all slots or could we simply compute `slotsForArgument[i]` in the next `for` loop?

Reply:
The reason I separated this loop is because all the slots need to be prepared before we Split (which the second loop may do).

---
In reply to: [243718357](https://github.com/dotnet/roslyn/pull/31978#discussion_r243718357) [](ancestors = 243718357)

Reply:
Ok, but `GetSlotsToMarkAsNotNullable` probably doesn't require a merged state.

---
In reply to: [243718639](https://github.com/dotnet/roslyn/pull/31978#discussion_r243718639) [](ancestors = 243718639,243718357)

Reply:
I was going to say it does. But you're correct.
This made me realize I'm missing a call to `Normalize` after `GetSlotsToMarkAsNotNullable`. This `Normalize` must happen in merged state.